lsp-mode.el

```diff
@@ -6734,28 +6734,42 @@ should return the command to start the LS server."
   "Create LSP stdio connection named name.
 LOCAL-COMMAND is either list of strings, string or function which
 returns the command to execute."
+  ;; 2.5.0-pre (as built from native-comp branch before M Albinus released tramp-2.5)
+  ;; worked fine
+  (defvar tramp-version)
+  (defvar tramp-connection-properties)
+  (when (version< tramp-version "2.5.0-pre")
+    (lsp-warn
+     "Your tramp version - %s - might fail to work with remote LSP. Update to version 2.5 or greater (available on elpa)"
+     tramp-version))
+  ;; Force a direct asynchronous process.
+  (add-to-list 'tramp-connection-properties
+               (list (regexp-quote (file-remote-p default-directory))
```

Comment:
I don't think this is right. `lsp-tramp-connection` is run when the user defines the lsp client, which may not be in a remote file (in which case the user gets an error). Plus, this is adding the entire directory to `tramp-connection-properties`, so it will grow forever as the user connects to different hosts. Don't we just need to add `("ssh:/" "tramp-direct-async" t)` to the list)?

Reply:
I think that like comes from the tramp maintainer. I am not 100% sure how significant it is.

Reply:
Trying this out, I can confirm that this errors if you're defining a tramp connection with elisp code in an indirect buffer (that is, one without an associated file name). I think the add-to-list here should happen inside the :connect lambda.

Reply:
It might not even be useful in this context.
See [the tramp source code](https://github.com/emacs-mirror/emacs/blob/1ff02c53b5f21dfcaff1794f3eb2f42fb6160dae/lisp/net/tramp.el#L4077)
`:buffer` and `:stderr` will be defined for this `make-process`, but those seem to prevent using `direct-async`.